sale_procurement_amendment/models/sale_order_line.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.tools import float_compare
+
+
+class SaleOrderLine(models.Model):
+
+    _inherit = 'sale.order.line'
+
+    to_be_procured = fields.Boolean(
+        compute='_compute_to_be_procured',
+        readonly=True,
+    )
+    pickings_in_progress = fields.Boolean(
+        compute='_compute_pickings_in_progress'
+    )
+
+    @api.multi
+    @api.depends('order_id.picking_ids.can_be_amended')
+    def _compute_pickings_in_progress(self):
+        """
+        Compute the picking in progress. That depends on picking
+        'can_be_amended' state. If one is not in the 'can_be_amended' state,
+        we are in the 'pickings_in_progress' situation.
+        :return:
+        """
+        for line in self.filtered(
+                lambda l: any(not picking.can_be_amended
+                              for picking in l.order_id.picking_ids)):
+            line.pickings_in_progress = True
+
+    @api.multi
+    @api.depends(
+        'procurement_ids',
+        'procurement_ids.state',
+        'procurement_ids.product_qty')
+    def _compute_to_be_procured(self):
```

Comment:
Should this depend on `product_uom_qty`, or is it not needed?

Reply:
Of course!